### ISSUE TITLE
tfvar files not accepted

### ISSUE THREAD

**Issue by Istyatur** [deliveredtechnologies/terraform-maven#132]
Expected: tf:apply, tf:destroy, and tf:plan should accept .tfvars files as input using the tfVarFiles property as if using the -var-file flag.

Result: the tfVarFiles does nothing. Using varFiles as the property name will produce a error and print terraform command usage for tf:apply and tf:plan.

Using terraform 0.12.9 on a RHEL 7 VirtualBox VM

**Comment by Clayton7510** [comment]
I'll check it out. Thanks for issue.

**Comment by vinaymbhoj** [comment]
Could you please suggest me how can I use the .tfvars files , is it possible to use it or not with current version ?

**Comment by Istyatur** [comment]
AFAIK it is not possible in the current version. I believe the bug can be traced back to 
the files in `tf-build-tools/tf-cmd-api/src/main/java/com/deliveredtechnologies/terraform/api/`; several files have the following data at the top

```
  enum TerraformApplyParam {
    tfVars("var"),
    varFiles("var_file"),
```
so terraform gets '-var_file' instead of '-var-file' 

but just changing var_file to var-file starts failing junit's and i never went further.

**Comment by Clayton7510** [comment]
I’ll drop an example in shortly. Probably this weekend.

**Comment by vinaymbhoj** [comment]
Thank you Clayton .

Regards
Vinay

**Comment by Clayton7510** [comment]
Looks like this is a bug. Working on a fix now.

**Comment by Clayton7510** [comment]
From the examples/tf-s3 root, the following command can be used to specify a tfvars file.
```mvn tf:apply -DtfVarFiles="$(pwd)/src/main/tf/s3/tfvars/dev.tfvars"```
You can also use the tfVarsFiles property with destroy and plan.
If you want to see the actual terraform command being executed, add the -X flag at the end for debug output.

**Comment by vinaymbhoj** [comment]
Thanks for fix. I trying to use this plugin on AZ. But till i am facing the same issue and not able to specify the tfvar file . 

mvn tf:plan -DtfVarFiles="src/main/tf/tfvars/bdd.tfvars" -X


 [DEBUG] Configuring mojo com.deliveredtechnologies:tf-maven-plugin:0.8-SNAPSHOT:plan from plugin realm ClassRealm[plugin>com.deliveredtechnologies:tf-maven-plugin:0.8-SNAPSHOT, parent: sun.misc.Launcher$AppClas
sLoader@4e25154f]
[DEBUG] Configuring mojo 'com.deliveredtechnologies:tf-maven-plugin:0.8-SNAPSHOT:plan' with basic configurator -->
[DEBUG] -- end configuration --
[DEBUG] *** directory: E:\customerAllocationInfra\customerAllocationInfra\src\main\tf\bdd ***
[DEBUG] *** command: terraform plan -input=false  ***

Error: No value for required variable

  on variables.tf line 4:
   4: variable "resource_group" {

The root module input variable "resource_group" is not set, and has no default
value. Use a -var or -var-file command line argument to provide a value for
this variable.


Error: No value for required variable

  on variables.tf line 12:
  12: variable "cosmosdb" {

The root module input variable "cosmosdb" is not set, and has no default
value. Use a -var or -var-file command line argument to provide a value for
this variable.

**Comment by Clayton7510** [comment]
What are you using for your terminal? Terraform 0.12 has a known bug using the Windows command line terminal. Instead, I recommend using GitBash, which you can set the plugin to fork to by setting the SHELL_PATH system environment variable. https://github.com/deliveredtechnologies/terraform-maven/blob/develop/README.md#windows-operating-systems

Also, use the full path, not a relative path. I noticed that Terraform 0.12 had difficulty us in relative paths for tfvar files.

**Comment by vinaymbhoj** [comment]
Ok. Have used git bash with file full path as well but the result is the same with no luck . Also build is failing with test failure however have build it by skipping the tests.

**Comment by Clayton7510** [comment]
Sorry, it took a bit to respond. But I was thinking a bit more about your issue. I think one issue is that the configurations included in the examples/tf-s3 project were for Terraform v0.11 and you are using v0.12. I updated those examples and their corresponding tests in the [bug/issue-132-var-files branch](https://github.com/deliveredtechnologies/terraform-maven/tree/bug/issue-132-var-files).

In that branch, from the examples/tf-s3 directory, if you type ```mvn test``` and assuming that your AWS creds are setup (and you have [GitBash shell configured](https://github.com/deliveredtechnologies/terraform-maven#windows-operating-systems)), you should see all of the tests pass.

Additionally, I have successfully tested the following using bash shell (non-Windows, since that's what I typically use) from the examples/tf-s3 project directory without making any updates to the 0.8-SNAPSHOT release. However, I did use the Terraform v0.12 compatible examples/tf-s3 project in the [bug/issue-132-var-files branch](https://github.com/deliveredtechnologies/terraform-maven/tree/bug/issue-132-var-files).

```mvn tf:init -DtfRootDir=s3```

```mvn tf:plan -DtfVarFiles="$(pwd)/src/main/tf/s3/tfvars/dev.tfvars" -DtfRootDir=s3 -DtfVars="is_versioned=true" -X```

```mvn tf:apply -DtfVarFiles="$(pwd)/src/main/tf/s3/tfvars/dev.tfvars" -DtfRootDir=s3 -DtfVars="is_versioned=true" -X```

```mvn tf:init -DtfRootDir=s3_replicated_src```

```mvn tf:plan -DtfVarFiles="$(pwd)/src/main/tf/s3_replicated_src/tfvars/dev.tfvars" -DtfRootDir=s3_replicated_src -DtfVars="destination_bucket_arn=arn:aws:s3:::bucket-b43f9e03ed4d6967" -X``` 
_note: the destination_bucket_arn came from the previous apply output_

```mvn tf:apply -DtfVarFiles="$(pwd)/src/main/tf/s3_replicated_src/tfvars/dev.tfvars" -DtfRootDir=s3_replicated_src -DtfVars="destination_bucket_arn=arn:aws:s3:::bucket-b43f9e03ed4d6967" -X```

```mvn tf:destroy -DtfVarFiles="$(pwd)/src/main/tf/s3_replicated_src/tfvars/dev.tfvars" -DtfRootDir=s3_replicated_src -DtfVars="destination_bucket_arn=arn:aws:s3:::bucket-b43f9e03ed4d6967" -X```

```mvn tf:destroy -DtfVarFiles="$(pwd)/src/main/tf/s3/tfvars/dev.tfvars" -DtfRootDir=s3 -DtfVars="is_versioned=true" -X```

I'll spin up a Windows VM tomorrow and try the same stuff using GitBash just to confirm there's nothing weird going on that I am not seeing in bash shell.

But anyway, if your issue was non-compliant Terraform 0.12 configs, which would be the case for the example project(s) in the develop branch, then the examples/tf-s3 project in the [bug/issue-132-var-files branch](https://github.com/deliveredtechnologies/terraform-maven/tree/bug/issue-132-var-files) has been updated to work with Terraform v0.12.

**Comment by Clayton7510** [comment]
On Windows, it looks like the pathing is a little different. And maybe this is the other issue you are running into. Clearly, I have to find a way to hook up automated tests for multi-platform. And Travis doesn't make that super easy. Although, supposedly more support for Windows testing is on the way.

Anyway, this is the equivalent of the above on Windows command line. And of course, I set my SHELL_PATH system environment variable to my GitBash executable. And I ran the commands below from the examples/tf-s3 directory in the [bug/issue-132-var-files branch](https://github.com/deliveredtechnologies/terraform-maven/tree/bug/issue-132-var-files) on a new Windows 10 VM with Maven 3.6.1, Java 1.8 and Terraform v0.12.21 installed.

```mvn tf:init -DtfRootDir=s3```

```mvn tf:plan -DtfVarFiles="$(pwd)/tfvars/dev.tfvars" -DtfRootDir=s3 -DtfVars="is_versioned=true" -X```

```mvn tf:apply -DtfVarFiles="$(pwd)/tfvars/dev.tfvars" -DtfRootDir=s3 -DtfVars="is_versioned=true" -X```

```mvn tf:init -DtfRootDir=s3_replicated_src```

```mvn tf:plan -DtfVarFiles="$(pwd)/tfvars/dev.tfvars" -DtfRootDir=s3_replicated_src -DtfVars="destination_bucket_arn=arn:aws:s3:::bucket-68d095fa98fe7e41" -X```

```mvn tf:apply -DtfVarFiles="$(pwd)/tfvars/dev.tfvars" -DtfRootDir=s3_replicated_src -DtfVars="destination_bucket_arn=arn:aws:s3:::bucket-68d095fa98fe7e41" -X```

```mvn tf:destroy -DtfVarFiles="$(pwd)/tfvars/dev.tfvars" -DtfRootDir=s3_replicated_src -DtfVars="destination_bucket_arn=arn:aws:s3:::bucket-68d095fa98fe7e41" -X```

```mvn tf:destroy -DtfVarFiles="$(pwd)/tfvars/dev.tfvars" -DtfRootDir=s3 -DtfVars="is_versioned=true" -X```

Did that help?

**Comment by Clayton7510** [comment]
This is fixed in v0.8 and v0.8.1
